os/net/lwip/src/api/sockets.c

```diff
@@ -453,31 +468,36 @@ struct lwip_sock *trycheck_selwait_socket(int s)
  */
 int alloc_socket(struct netconn *newconn, int accepted)
 {
-	int i;
+	int idx;
+	struct socketlist *list;
+	struct lwip_sock *sock;
+
+	list = sched_getsockets();
+	if (!list) {
+		ndbg("could not get the socketlist\n");
+		return -1;
+	}
+
 	SYS_ARCH_DECL_PROTECT(lev);
+	sock = (struct lwip_sock *)kmm_zalloc(sizeof(struct lwip_sock));
+	if (!sock) {
+		ndbg("memory is not sufficient for sockets\n");
+		return -1;
```

Comment:
please set ERROR or ENOMEM instead of -1

Reply:
it will not matter much because when it will return -1, there it set ENFILE as error.